module-moderation/src/main/kotlin/org/quiltmc/community/cozy/modules/moderation/ModerationExtension.kt

```diff
@@ -0,0 +1,123 @@
+import dev.kord.core.entity.channel.TextChannel
+import org.quiltmc.community.cozy.modules.moderation.config.ModerationConfig
+
+/**
+ * Moderation, extension, provides different moderation related tools.
+ *
+ * Currently includes:
+ * - Slowmode command
+ */
+public class ModerationExtension(
+    private val config: ModerationConfig
+) : Extension() {
+    override val name: String = ModerationPlugin.id
+
+    override suspend fun setup() {
+        ephemeralSlashCommand {
+            name = "slowmode"
+            description = "Manage slowmode of the current channel"
+
+            check { anyGuild() }
+            check { isNotInThread() }
+
+            config.getCommandChecks().forEach(::check)
+
+            ephemeralSubCommand {
+                name = "get"
+                description = "Get the slowmode of the channel"
+
+                action {
+                    respond {
+                        content = "Slowmode is currently " +
+                                "${channel.asChannel().data.rateLimitPerUser.value ?: 0} second(s)."
+                    }
+                }
+            }
+
+            ephemeralSubCommand {
+                name = "reset"
+                description = "Reset the slowmode of the channel back to 0"
+
+                action {
+                    val channel = channel.asChannel() as TextChannel
+
+                    channel.edit {
+                        rateLimitPerUser = 0
+                    }
+
+                    respond {
+                        content = "Slowmode reset."
+                    }
+                }
+            }
+
+            ephemeralSubCommand(::SlowmodeEditArguments) {
+                name = "set"
+                description = "Set the slowmode of the channel"
+
+                action {
+                    val channel = channel.asChannel() as TextChannel
+
+                    channel.edit {
+                        rateLimitPerUser = arguments.duration
+                    }
+
+                    config.getLoggingChannelOrNull(guild!!.asGuild())?.createEmbed {
+                        title = "Slowmode changed"
+                        description = "Set to ${arguments.duration} second(s)."
+                        color = DISCORD_BLURPLE
+
+                        field {
+                            inline = true
+                            name = "Channel"
+                            value = channel.mention
+                        }
+
+                        field {
+                            inline = true
+                            name = "User"
+                            value = user.mention
+                        }
+                    }
+
+                    respond {
+                        content = "Slowmode set to ${arguments.duration} second(s)."
+                    }
+                }
+            }
+        }
+    }
+
+    public inner class SlowmodeEditArguments : Arguments() {
+        public val duration: Int by int {
```

Comment:
Would it be worth making this use a duration converter? I'm not sure if it's truly that useful, but being able to specify `30m` or `2h` might be nice.

The limit is 6 hours, by the way, if you want to add that to the validator.

Reply:
Done in 787dea0